spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/client/loadbalancer/impl/scoped/RoundRobinScopedLoadBalancer.java

```diff
@@ -0,0 +1,65 @@
+package org.springframework.cloud.client.loadbalancer.impl.scoped;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.client.loadbalancer.impl.support.LoadBalancerClientFactory;
+
+import static org.springframework.util.CollectionUtils.isEmpty;
+
+/**
+ * @author Spencer Gibb
+ */
+public class RoundRobinScopedLoadBalancer implements ScopedLoadBalancer {
```

Comment:
looks like c&p for some of these things.. if meaty code, maybe an internal utility that they share. otherwise, keep an eye on it.